offixsdk/src/main/java/org/aerogear/offixsdk/ClientGenerator.kt

```diff
@@ -0,0 +1,139 @@
+package org.aerogear.offixsdk
+
+import android.content.Context
+import android.util.Log
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.cache.normalized.CacheKey
+import com.apollographql.apollo.cache.normalized.CacheKeyResolver
+import com.apollographql.apollo.cache.normalized.lru.EvictionPolicy
+import com.apollographql.apollo.cache.normalized.lru.LruNormalizedCacheFactory
+import com.apollographql.apollo.cache.normalized.sql.ApolloSqlHelper
+import com.apollographql.apollo.cache.normalized.sql.SqlNormalizedCacheFactory
+import com.apollographql.apollo.subscription.WebSocketSubscriptionTransport
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import org.aerogear.offixsdk.Interceptor.LoggingInterceptor
+import java.nio.charset.Charset
+
+/*  ClientGenerator class will help the user in creating an apollo client.
+    The user passes in the server url to which the request has to be sent.
+*/
+
+class ClientGenerator(val serverUrl: String) {
+
+    private val TAG = javaClass.simpleName
+    private val applicationContext: Context = getContext()
+    val SQL_CACHE_NAME = "oclientDB"
+    private var apClient: ApolloClient? = null
```

Comment:
Use lateinit var and initialize the variable asap instead of using nullable types